src/sentry/workflow_engine/migrations/0015_create_rule_lookup_tables.py

```diff
@@ -0,0 +1,155 @@
+        migrations.CreateModel(
+            name="AlertRuleTriggerDataCondition",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("date_updated", models.DateTimeField(auto_now=True)),
+                ("date_added", models.DateTimeField(auto_now_add=True)),
+                (
+                    "alert_rule_trigger",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.alertruletrigger"
+                    ),
+                ),
+                (
+                    "data_condition",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="workflow_engine.datacondition",
+                    ),
+                ),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        migrations.CreateModel(
+            name="AlertRuleDetector",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("date_updated", models.DateTimeField(auto_now=True)),
+                ("date_added", models.DateTimeField(auto_now_add=True)),
+                (
+                    "alert_rule",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.alertrule",
+                    ),
+                ),
+                (
+                    "detector",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="workflow_engine.detector"
+                    ),
+                ),
+                (
+                    "rule",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.rule"
+                    ),
+                ),
+            ],
+            options={
+                "db_table": "workflow_engine_alertruledetector",
+                "constraints": [
+                    models.CheckConstraint(
+                        condition=models.Q(
+                            models.Q(("alert_rule__isnull", True), ("rule__isnull", False)),
+                            models.Q(("alert_rule__isnull", False), ("rule__isnull", True)),
+                            _connector="OR",
+                        ),
```

Comment:
```suggestion
                        condition=Q(rule__isnull=False, alert_rule__isnull=True)
                          | Q(rule__isnull=True, alert_rule__isnull=False),
```

Could align this with the other index definitions.